Tests/RoutingKitTests/RouterTests.swift

```diff
@@ -265,28 +284,50 @@ final class RouterTests: XCTestCase {
             case .catchall:
                 break
             default:
-                XCTFail("pathComponentInterpolation \(pathComponentInterpolation) is not .catchall")
+                Issue.record("pathComponentInterpolation \(pathComponentInterpolation) is not .catchall")
             }
         }
     }
 
-    func testParameterNamesFetch() throws {
-        let router = TrieRouter<Int>()
-        router.register(42, at: ["foo", ":bar", ":baz", ":bam"])
-        router.register(24, at: ["bar", ":bar", "**"])
+    @Test func parameterNamesFetch() throws {
+        var builder = TrieRouterBuilder<Int>()
+        builder.register(42, at: ["foo", ":bar", ":baz", ":bam"])
+        builder.register(24, at: ["bar", ":bar", "**"])
+        let router = builder.build()
 
         var params1 = Parameters()
-        XCTAssertNil(router.route(path: ["foo"], parameters: &params1))
-        XCTAssertTrue(params1.getCatchall().isEmpty)
+        #expect(router.route(path: ["foo"], parameters: &params1) == nil)
+        #expect(params1.getCatchall().isEmpty)
 
         var params2 = Parameters()
-        XCTAssertEqual(router.route(path: ["foo", "a", "b", "c"], parameters: &params2), 42)
-        XCTAssertEqual(Set(params2.allNames), ["bar", "baz", "bam"])  // Set will compare equal regardless of ordering
-        XCTAssertTrue(params2.getCatchall().isEmpty)
+        #expect(router.route(path: ["foo", "a", "b", "c"], parameters: &params2) == 42)
+        #expect(Set(params2.allNames) == ["bar", "baz", "bam"])  // Set will compare equal regardless of ordering
+        #expect(params2.getCatchall().isEmpty)
 
         var params3 = Parameters()
-        XCTAssertEqual(router.route(path: ["bar", "baz", "bam"], parameters: &params3), 24)
-        XCTAssertEqual(Set(params3.allNames), ["bar"])
-        XCTAssertEqual(params3.getCatchall(), ["bam"])
+        #expect(router.route(path: ["bar", "baz", "bam"], parameters: &params3) == 24)
+        #expect(Set(params3.allNames) == ["bar"])
+        #expect(params3.getCatchall() == ["bam"])
+    }
+
+    @Test func testPartial() throws {
+        var routerBuilder = TrieRouterBuilder<Int>()
+        routerBuilder.register(42, at: ["test", ":{my-file}.json"])
```

Comment:
Let's add some more tests to test things like a partial parameter only at the start, only at the end, failures for unclosed parameters etc